panoptes_aggregation/reducers/subject_difficulty_reducer.py

```diff
@@ -0,0 +1,38 @@
+'''
+Subject Gold Standard Reducer for difficulty calculation
+---------------------------------------------------------
+This module provides functions to reduce the gold standard task extracts
+to determine a `difficulty' score per subject (defined as the fraction
+of succesful classification by all users for that subject)
+
+See also: tess_gold_standard_reducer
+'''
+from .reducer_wrapper import reducer_wrapper
+import numpy as np
+
+
+@reducer_wrapper()
+def subject_difficulty_reducer(extracts):
+    '''
+    Calculate the difficulty of a gold standard TESS subject
+
+    Parameters
+    ----------
+    extracts : list
+        The list of extracted data including the feedback metadata
+
+    Returns
+    -------
+    output : dict
+        A dictinary with one key `difficulty` that is a list with the fraction of volunteers who
+        successfully found each gold standard entry in a subject.
+    '''
+
+    feedback_data = [extracti['feedback'] for extracti in extracts if extracti['feedback'] is not None]
+
+    output = {}
+    if len(feedback_data) > 0:
+        success = np.asarray([feedback['success'] for feedback in feedback_data])
+        difficulty = np.sum(success, axis=0) / len(success)
+        output['difficulty'] = difficulty.tolist()
```

Comment:
You don't need to explicitly cast to the array data type, the numpy function can also just take in a list of lists and convert to a 2d array on their own.  Example:

```python
import numpy as np
a = [[1, 2, 3], [4, 5, 6]]
difficulty = np.mean(a, axis=0).tolist()
```

I propose this code can be written as:

```python
success = [extract['feedback']['sucess'] for extract in extracts if 'feedback' in extract]

output = {}
if len(success) > 0:
    output['difficulty'] = np.mean(sucess, axis=0).tolist()
```

This reduces the number of times the data is looped over, and also make the intention of the code easier to read. Just double-check that the proposed list comprehension actually works as expected...